src/lib/ScannerCommand.ts

```diff
@@ -2,9 +2,35 @@ import {SfdxCommand} from '@salesforce/command';
 import {CategoryFilter, LanguageFilter, RuleFilter, RulesetFilter, RulenameFilter, EngineFilter} from './RuleFilter';
 import {uxEvents, EVENTS} from './ScannerEvents';
 import {stringArrayTypeGuard} from './util/Utils';
+import {AnyJson} from '@salesforce/ts-types';
+
+import {Messages} from '@salesforce/core';
+
+// Initialize Messages with the current plugin directory
+Messages.importMessagesDirectory(__dirname);
+const commonMessages = Messages.loadMessages('@salesforce/sfdx-scanner', 'common');
+
 
 export abstract class ScannerCommand extends SfdxCommand {
 
+	public async run(): Promise<AnyJson> {
+		this.runCommonSteps();
+		return await this.runInternal();
+	}
+
+	/**
+	 * Command's should implement this method to add their
+	 * working steps.
+	 */
+	abstract runInternal(): Promise<AnyJson>;
+
+	/**
+	 * Common steps that should be run before every command
+	 */
+	protected runCommonSteps(): void {
```

Comment:
A place to put common steps so that we don't have to modify each command every time we change things.

Reply:
This being its own method feels excessive. You could just do these things in `run()` before invoking `runInternal()`

Reply:
`runCommonSteps()` behaves as a hook that would continue to keep `run()` more readable as the common steps expand.
Also, we may have a subgroup of commands that require to do a variation of the common steps and override it to their convenience. For example, in an abstract class like ScannerRunCommand.
In near future, if we decide to remove the survey link, I'd still recommend leaving the hook with a marker comment on what can possibly be added there.